packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,181 @@ where
+    }
+
+    #[test]
+    fn multiple_copies_best() {
+        // The two 9s are the possible winners
+        let population = vec![5, 8, 9, 6, 3, 2, 0, 9]
+            .into_iter()
+            .enumerate()
+            .map(|(id, e)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from([e])))
+            .collect::<Vec<_>>();
+
+        let lexicase = Lexicase::new(1);
+        let mut rng = rand::thread_rng();
+
+        let selected = *lexicase.select(&population, &mut rng).unwrap().genome();
+        assert!(
+            selected == 2usize || selected == 7usize,
+            "genome {selected} should have been 2 or 7"
+        );
+    }
+
+    #[test]
+    fn single_best_multiple_errors() {
+        // [9, 8] is the sole winner
+        let population = vec![[5, 3], [8, 2], [9, 8], [6, 2], [3, 8], [2, 8], [0, 6]]
+            .into_iter()
+            .enumerate()
+            .map(|(id, es)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from(es)))
+            .collect::<Vec<_>>();
+
+        let lexicase = Lexicase::new(2);
+        let mut rng = rand::thread_rng();
+
+        assert_eq!(&2, lexicase.select(&population, &mut rng).unwrap().genome());
+    }
+
+    #[test]
+    fn multiple_best_multiple_errors() {
+        let population = vec![
+            [5, 3],
+            [8, 2],
+            [9, 8], // A possible winner
+            [6, 2],
+            [3, 8],
+            [2, 8],
+            [0, 6],
+            [7, 9], // A possible winner
+        ]
+        .into_iter()
+        .enumerate()
+        .map(|(id, es)| EcIndividual::<usize, TestResults<i32>>::new(id, TestResults::from(es)))
+        .collect::<Vec<_>>();
+
+        let lexicase = Lexicase::new(2);
+        let mut rng = rand::thread_rng();
+
+        let selected = *lexicase.select(&population, &mut rng).unwrap().genome();
+        assert!(
+            selected == 2usize || selected == 7usize,
+            "genome {selected} should have been 2 or 7"
+        );
+    }
+
+    // A name for this type to simplify things in the `proptest` test
+    // below.
+    type TestIndividual = EcIndividual<String, TestResults<u16>>;
+
+    // This test uses `proptest` to generate a random set of between 1 and 20
+    // vectors of two scores, and then converts those into a vector of
+    // `EcIndividual`s (i.e., a population). I then determine the largest of the
+    // first values (i.e., the highest score on the first test case), and then the
+    // largest of the second values. I make a new set of results that is those
+    // highest values with one (randomly chosen by proptest) incremented by 1.
+    // This new result is guaranteed to be "better" than any other in the list,
+    // so it should be what is selected after I add it to the population.
+    //
+    // There are several helper functions that exist just to support the somewhat
+    // complex logic of this test.
+    #[test_strategy::proptest]
```

Comment:
everywhere else we just import `test_strategy::proptest` - do we want to do that here as well?

Reply:
Good catch. I imported `proptest::prelude::*` (which is what they do in the `proptest` book), and that created a conflict that I dealt with by qualifying `test_strategy::proptest` here.

I've replaced the glob import with several specific imports, which avoids that conflict and we can just use `#[proptest]` here again.